gradio/blocks.py

```diff
@@ -743,7 +743,19 @@ def __call__(self, *inputs, fn_index: int = 0):
         Parameters:
         *inputs: the parameters to pass to the function
         fn_index: the index of the function to call (defaults to 0, which for Interfaces, is the default prediction function)
+        api_name: The api_name of the dependency to call. Will take precedence over fn_index.
         """
+        if api_name is not None:
+            fn_index = next(
+                (
+                    i
+                    for i, d in enumerate(self.dependencies)
+                    if d.get("api_name") == api_name
+                ),
+                None,
+            )
```

Comment:
Just out of curiosity, what is the advantage of this next(generator) approach over doing list.index()? I find the latter more readable

Reply:
I like it cause in the case of the api_name not being there, it will return None instead of raising an exception.

Are you suggesting?

```python
try:
    fn_index = [d.get('api_name') for d in self.dependecies].index(api_name)
except ValueError:
    raise (....)    
```    

Happy to switch implementation!

Reply:
Yes that's what I was suggesting, but either way is fine! It probably speaks more to my personal comfort that I prefer the second over the first, I don't think there is one that's actually more pythonic

Reply:
Yea I agree both are equivalently pythonic (or non pythonic 😆 ) 

I personally prefer if the failure cause of a function is None (or some value like that) as opposed to raising an exception 🤷